pyam/netcdf.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import datetime as dt
+import pandas as pd
+
+try:
+    import xarray as xr
+    
+    HAS_XARRAY = True
+except ModuleNotFoundError:
+    xr = None
+    HAS_XARRAY = False
+from pyam.core import IamDataFrame
+from pyam.utils import META_IDX, IAMC_IDX
+
+
+def read_netcdf(path):
+    """Read timeseries or year-based data and meta-indicators from netCDF
```

Comment:
Yearly data is also called timeseries, no need to differentiate here
```suggestion
    """Read timeseries data and meta indicators from a netCDF file
```